ksql-engine/src/test/java/io/confluent/ksql/EndToEndEngineTestUtil.java

```diff
@@ -520,20 +537,115 @@ private void handleException(final RuntimeException e) {
     }
   }
 
-  private static TopologyTestDriver buildStreamsTopology(final Query query,
-                                                         final KsqlEngine ksqlEngine,
-                                                         final KsqlConfig ksqlConfig,
-                                                         final Properties streamsProperties) {
-    final List<QueryMetadata> queries = new ArrayList<>();
-    query.statements().forEach(
-        q -> queries.addAll(
-            ksqlEngine.buildMultipleQueries(q, ksqlConfig, query.properties()))
-    );
-    return new TopologyTestDriver(queries.get(queries.size() - 1).getTopology(),
+
+  static void writeExpectedTopologyFiles(final String topologyDir) throws IOException {
+    final List<Query> queryList = QueryTranslationTest.buildQueryList().stream()
+                                                                       .filter(q -> q.expectedException.matchers.isEmpty())
+                                                                       .collect(Collectors.toList());
+
+    final Random randomPort = new Random();
+    queryList.forEach(query -> {
+      final Map<String, Object> config = getConfigs(null);
+      config.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:" + randomPort.nextInt(4000));
+      final KsqlConfig ksqlConfig = new KsqlConfig(config);
+      final KsqlEngine ksqlEngine = getKsqlEngine(new MockSchemaRegistryClient(), ksqlConfig);
+      final Topology topology = getStreamsTopology(query, ksqlEngine, ksqlConfig);
+      writeExpectedTopologyFile(query.name, topology, topologyDir);
```

Comment:
Since we use the query.name as the file name, we may need to make sure that 1) these query names would not be changed in the test classes, 2) when tests / queries are removed, their corresponding topology files should also be removed from the corresponding directory.

Reply:
Changing the query name will cause a test failure and require a rebuild of the expected topologies for that version.  I'll update with some comments to that effect.

However, dropping a test or a query should not cause any issue as that test or query just won't get retrieved from the map, but it's worth test.